library/tiqr/tests/Tiqr_UserSecretStorageTest.php

```diff
@@ -29,6 +58,159 @@ public function test_it_can_create_user_sercret_storage($type, $expectedInstance
         $this->assertInstanceOf(Tiqr_UserSecretStorage_Interface::class, $userSecretStorage);
     }
 
+    /**
+     * Setup a user secret storage instance of file or pdo type in a temporary directory
+     * During a test, the user secret storage instance can be recreated with the same type and options, and it will return
+     * the same data
+     * @param string $type: file or pdo
+     * @param array $encryption_options: encryption/decryption options
+     * @param array $raw_user_secret: raw user secret data, array(userid=>secret), this is used to initialize the storage
+     *                                directly, without using the setSecret() method so it bypasses the encryption of the
+     *                                secret and the prefixing of the secret with the encryption type
+     *
+     * @return Tiqr_UserSecretStorage_Interface
+     */
+    public function setup_userSecretStorage(string $type, array $encryption_options = array(), array $raw_user_secret = array()) : Tiqr_UserSecretStorage_Interface
+    {
+        switch ($type) {
+            case 'pdo':
+                $encryption_options['dsn'] = "sqlite://{$this->temp_dir}/db.sqlite";
+                $encryption_options['username'] = 'user';
+                $encryption_options['password'] = 'secret';
+
+                $pdo = new PDO($encryption_options['dsn'],
+                    null,
+                    null,
+                    [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]
+                );
+                // If required, init schema
+                $pdo->exec("CREATE TABLE IF NOT EXISTS tiqrusersecret (userid varchar(255) PRIMARY KEY, secret varchar(255));");
+
+                foreach ($raw_user_secret as $user => $secret) {
+                    $s = $pdo->prepare("REPLACE INTO tiqrusersecret (userid, secret) VALUES (:user, :secret);");
+                    $s->bindParam(':user', $user);
+                    $s->bindParam(':secret', $secret);
+                    $s->execute();
+                }
+                break;
+
+            case 'file':
+                $encryption_options['path'] = $this->temp_dir;
+
+                foreach ($raw_user_secret as $user => $secret) {
+                    $filename = $this->temp_dir . '/' . $user;
+                    file_put_contents($filename . '.json', json_encode(['secret' => $secret]));
+                }
+                break;
+
+            default:
+                throw new RuntimeException("Unknown type: $type");
+        }
+
+        return Tiqr_UserSecretStorage::getSecretStorage(
+            $type,
+            Mockery::mock(LoggerInterface::class)->shouldIgnoreMissing(),
+            $encryption_options
+        );
+    }
+
+    /**
+     * @dataProvider provideUserSecretStorageTypes
+     */
+    public function test_it_can_use_user_secret_storage_with_openssl_encryption(string $type)
```

Comment:
This method might be a good candidate to deck out with a dataProvider that broadens the test coverage of this method to test different ciphers.

Reply:
What I want to test here is that the openssl encryption class is instantiated correctly, and that the options array is passed to it correctly.

All the different ciphers are tested here:
https://github.com/Tiqr/tiqr-server-libphp/blob/cb45348483d2ae24ce3db1343f00ccf40acf92ab/library/tiqr/tests/Tiqr_UserSecretStorage_Encryption_OpenSSLTest.php#L47-L80